File: baekjoon/brute-force/python/BOJ21275.py
from sys import stdin  
readline = stdin.readline

A, B = readline().split()

def solve():
    ans = []

    for i in range(2, 37):
        a = conversion_to_ten(A, i)
        if a == -1:
            continue

        for j in range(2, 37):

            b = conversion_to_ten(B, j)

            if b == -1 or i == j:
                continue

            if a == b:
                if not ans:
                    ans = [a, i, j]
                else:
                    print("Multiple")
                    return
    
    if not ans:
        print("Impossible")
    else:
        print(*ans)

def conversion_to_ten(num, N):
    result = 0

    for i, n in enumerate(num[::-1]):
        if str.isdigit(n):
            current = int(n)
        else:
            current = (10 + ord(n) - ord('a'))

        if current >= N:
            return -1
        
        result += current * N**i

    if result >= 2**63:
        return -1

    return result

solve()

File: baekjoon/brute-force/java/BOJ21275.java
import java.io.*;
import java.util.*;

public class BOJ21275 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());
        String a = st.nextToken();
        String b = st.nextToken();
        long[] result = {-1, -1, -1};

        for (int i=2;i<=36;i++) {
            long A = convertToTen(a, i);

            for (int j=2;j<=36;j++) {
                if (i == j) continue;
                long B = convertToTen(b, j);

                if (A == B && A != -1 && A != Long.MAX_VALUE) {
                    if (result[0] != -1) {
                        System.out.println("Multiple");
                        return;
                    } else { 
                        result = new long[] {A, i, j};
                    }
                }
            }
        }
        if (result[0] == -1) {
            System.out.println("Impossible");
        } else {
            System.out.println(result[0] +" " + result[1] + " " + result[2]);
        }

    }

    private static long convertToTen(String num, int n) {

        long result = 0;
        int len = num.length() - 1;

        for (int i = len; i >= 0; i--) {
            char ch = num.charAt(i);
            int current = 0;
            if (Character.isDigit(ch)) {
                current = (ch - '0');
            } else {
                current = (10 + ch - 'a');
            }

            if (current >= n)
                return -1;

            result += current * Math.pow(n, len - i);
        }

        return result;
    }
}
